src/theory/strings/theory_strings.h

```diff
@@ -580,7 +580,13 @@ class TheoryStrings : public Theory {
   void doPendingFacts();
   void doPendingLemmas();
   bool hasProcessed();
+  /** 
+   * Adds equality a = b to the vector exp if a and b are distinct. Checks 
+   * that a = b holds in this context, throws an assertion failure via Assert
+   * if not.
```

Comment:
I would just say "a = b must hold in this context" because it is only actually checked in debug mode and this description makes it seem like it is always checked and the exception could be caught by the callee.